pkg/utils/k8s.go

```diff
@@ -139,25 +139,41 @@ func ExecToPod(config *rest.Config, command []string, containerName, podName, na
 		Stdin:     stdin != nil,
 		Stdout:    true,
 		Stderr:    true,
-		TTY:       false,
 	}, parameterCodec)
 
 	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
 	if err != nil {
-		return "", "", errors.WithStack(fmt.Errorf("error while creating Executor: %v", err))
+		return errors.WithStack(fmt.Errorf("error creating Executor: %v", err))
 	}
 
-	var stdout, stderr bytes.Buffer
-	if err = exec.Stream(remotecommand.StreamOptions{
+	var stderr bytes.Buffer
+	l := &LogBuffer{}
+	err = exec.Stream(remotecommand.StreamOptions{
 		Stdin:  stdin,
-		Stdout: &stdout,
+		Stdout: l,
 		Stderr: &stderr,
-		Tty:    false,
-	}); err != nil {
-		return stdout.String(), stderr.String(), errors.WithStack(fmt.Errorf("error in Stream: %v", err))
+	})
+	if err != nil {
+		if stderr.String() != "" {
+			// cli returned an error
+			return fmt.Errorf(stderr.String())
+		}
+		// error from the exec cmd
+		return err
 	}
+	return nil
+}
+
+// LogBuffer is a custom implementation of the io.Writer interface that logs
+// what's on the buffer on every Write thus printing its contents as close to
+// "real time" as possible.
+// Using this method we don't have to wai for CLI commands to finish before
+// being able to print the buffers (stdout/err).
+type LogBuffer struct{}
 
-	return stdout.String(), stderr.String(), nil
+func (l *LogBuffer) Write(p []byte) (int, error) {
+	fmt.Print(string(p))
```

Comment:
If we blindly print here then we end up with stdout from etcd endpoint validation check (OK, foo, bar etc...):

![image](https://user-images.githubusercontent.com/41484746/217243956-3a12cca7-273c-4462-a96f-f126f5aa7572.png)